code/spaceship/CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+
+# Set up the project.
+cmake_minimum_required( VERSION  3.12 )
+project( spaceship LANGUAGES CXX )
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# Set up the compilation environment.
+include( "${CMAKE_CURRENT_SOURCE_DIR}/../CompilerSettings.cmake" )
+include( "${CMAKE_CURRENT_SOURCE_DIR}/../SolutionTarget.cmake" )
```

Comment:
The problem with this is, that you set `CMAKE_CXX_STANDARD` to 20, and then `CompilerSettings.cmake` sets it back to 17. You can either swap the lines like this:
```suggestion
# Set up the compilation environment.
include( "${CMAKE_CURRENT_SOURCE_DIR}/../CompilerSettings.cmake" )
include( "${CMAKE_CURRENT_SOURCE_DIR}/../SolutionTarget.cmake" )

set(CMAKE_CXX_STANDARD 20)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

Or just request C++20 directly on the target:
```cmake
add_executable( spaceship "spaceship.cpp" )
target_compile_features(spaceship PUBLIC cxx_std_20)
```

Both options should work.